File: api/app/Models/StockMovement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class StockMovement extends Model
{
    protected $fillable = [
        'stock_id', 'product_id', 'warehouse_id', 'color_id', 'size',
        'movement_type', 'quantity_before', 'quantity_change', 'quantity_after',
        'reason', 'reference', 'user_id',
    ];

    protected $casts = [
        'quantity_before'  => 'integer',
        'quantity_change'  => 'integer',
        'quantity_after'   => 'integer',
    ];

    public function stock(): BelongsTo     { return $this->belongsTo(Stock::class); }
    public function product(): BelongsTo   { return $this->belongsTo(Product::class); }
    public function warehouse(): BelongsTo { return $this->belongsTo(Warehouse::class); }
    public function color(): BelongsTo     { return $this->belongsTo(Color::class); }
    public function user(): BelongsTo      { return $this->belongsTo(User::class); }
}

File: api-dotnet/AdminHiitop.Api/Domain/Inventory/Entities/StockMovement.cs
using AdminHiitop.Api.Domain.Catalog.Entities;
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Inventory.Entities;

public sealed class StockMovement : AuditableEntity
{
    public int StockId { get; set; }
    public int ProductId { get; set; }
    public int WarehouseId { get; set; }
    public int? ColorId { get; set; }
    public string? Size { get; set; }
    public string MovementType { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public int PreviousQuantity { get; set; }
    public int NewQuantity { get; set; }
    public string? Reference { get; set; }
    public string? Notes { get; set; }
    public int? UserId { get; set; }

    public Stock Stock { get; set; } = null!;
    public Product Product { get; set; } = null!;
    public Warehouse Warehouse { get; set; } = null!;
    public Color? Color { get; set; }
    public User? User { get; set; }
}
